core/src/main/java/com/amplifyframework/devmenu/LogEntry.java

```diff
@@ -0,0 +1,131 @@
+    // The format for the log's date and time.
+    private static final String DATE_TIME_FORMAT = "yyyy-MM-dd HH:mm:ss.SSS";
+    private final LocalDateTime dateTime;
+    private final String namespace;
+    private final String message;
+    private final Throwable throwable;
+    private final LogLevel logLevel;
+
+    /**
+     * Creates a new LogEntry representing a log with the given time, tag, message,
+     * and throwable that was logged at the given level.
+     * @param dateTime the date and time of the log.
+     * @param namespace the namespace of the logger that emitted the log.
+     * @param message the message for the log.
+     * @param throwable the Throwable associated with the log.
+     * @param logLevel the level the log was logged at.
+     */
+    public LogEntry(@NonNull LocalDateTime dateTime, @Nullable String namespace, @Nullable String message,
+                    @Nullable Throwable throwable, @NonNull LogLevel logLevel) {
+        this.dateTime = Objects.requireNonNull(dateTime);
+        this.logLevel = Objects.requireNonNull(logLevel);
+        this.namespace = namespace;
+        this.message = message;
+        this.throwable = throwable;
+    }
+
+    /**
+     * Gets the date and time of the log.
+     * @return the date and time of the log.
+     */
+    public LocalDateTime getDateTime() {
+        return dateTime;
+    }
+
+    /**
+     * Gets the namespace of the logger that emitted the log.
+     * @return the namespace of the logger that emitted the log.
+     */
+    public String getNamespace() {
+        return namespace;
+    }
+
+    /**
+     * Gets the message for the log.
+     * @return the message for the log.
+     */
+    public String getMessage() {
+        return message;
+    }
+
+    /**
+     * Gets the throwable for the log.
+     * @return the throwable for the log, or null if there is no throwable for the log.
+     */
+    public Throwable getThrowable() {
+        return throwable;
+    }
+
+    /**
+     * Gets the level the log was logged at.
+     * @return the level the log was logged at.
+     */
+    public LogLevel getLogLevel() {
+        return logLevel;
+    }
+
+    @Override
+    public int compareTo(LogEntry logEntry) {
+        return getDateTime().compareTo(logEntry.getDateTime());
+    }
+
+    @Override
+    public boolean equals(Object object) {
+        if (!(object instanceof LogEntry)) {
+            return false;
+        }
+        LogEntry otherEntry = (LogEntry) object;
+        return dateTime.equals(otherEntry.getDateTime()) && logLevel.equals(otherEntry.getLogLevel())
+                && namespace.equals(otherEntry.getNamespace()) && message.equals(otherEntry.getMessage())
+                && throwable.equals(otherEntry.getThrowable());
+    }
+
+    @Override
+    public int hashCode() {
+        return dateTime.hashCode() + logLevel.hashCode() + namespace.hashCode() + message.hashCode()
+                + throwable.hashCode();
```

Comment:
For both `equals()` and `hashCode()`, you can have the IDE generate them for you. There's a standard recipe for `hashCode()` in Java which involves offsetting individual component hash codes with a prime.

In Android Studio, right click in the main document of the `LogEntry` class, and click "Generate..." then select "equals() and hashCode()." For "Template," I use "IntelliJ Default." The next page will ask you which values can be null or non-null, and you can follow along with what you put in your constructor params.